PlaintextContent not handled

Note: I practised some proper tomfoolery with my Signal Desktop database before this happened, so this may be a situation Signal isn't supposed to end up in. Reader discretion advised.

Using Whisperfish and Signal Desktop, I managed to have cross-device (sync?) messages sent as `PlaintextContent` type. Here's roughly how it went:

- Have Whisperfish open, linked Signal Desktop closed
- Send a message to self (or another contact)
- Open Signal Desktop
- &rarr; Lots of "A message from [my_name] couldn't be delivered" messages appear (one per sent message)
- None of the messages sent from WF to SD show up, the other direction works

Here's a snippet from Whisperfish log:

```
2023-03-25 17:46:52.917 [TRACE] libsignal_service::envelope: Envelope::decrypt: not encrypted
2023-03-25 17:46:52.917 [WARN] Unknown envelope type PlaintextContent
2023-03-25 17:46:52.918 [ERROR] Error opening envelope: InvalidFrameError { reason: "Envelope has unknown type PlaintextContent." }
```

After following the code for a bit, I found out that envelope type 8 is `PLAINTEXT_CONTENT` and that it is not matched in `libsignal-service/src/cipher.rs` function `decrypt(&mut self, envelope: Envelope)`, instead it falls to the default case:

```
_ => {
    // else
    return Err(ServiceError::InvalidFrameError {
        reason: format!(
            "Envelope has unknown type {:?}.",
            envelope.r#type()
        ),
    });
},
```

Signal logs look like this:

```
{"level":30,"time":"2023-03-25T18:05:44.194Z","msg":"MessageReceiver.decryptAndCacheBatch 1"}
{"level":30,"time":"2023-03-25T18:05:44.194Z","msg":"SignalProtocolStore.enterZone(decryptAndCacheBatch:MessageReceiver)"}
{"level":30,"time":"2023-03-25T18:05:44.195Z","msg":"queueing envelope [REDACTED]2ce.1 > [REDACTED]2ce 1679767534505 (REDACTED_SOMETHING)"}
{"level":30,"time":"2023-03-25T18:05:44.195Z","msg":"MessageReceiver.decryptEnvelope([REDACTED]2ce.1 > [REDACTED]2ce 1679767534505 (REDACTED_SOMETHING))"}
{"level":30,"time":"2023-03-25T18:05:44.196Z","msg":"decrypt/[REDACTED]2ce.1 > [REDACTED]2ce 1679767534505 (REDACTED_SOMETHING): ciphertext message"}
{"level":50,"time":"2023-03-25T18:05:44.196Z","msg":"queueEncryptedEnvelope error handling envelope [REDACTED]2ce.1 > [REDACTED]2ce 1679767534505 (REDACTED_SOMETHING) : LibSignalError: session with [REDACTED]2ce.1 not found\n    at new LibSignalErrorBase ([REDACTED]/resources/app.asar/node_modules/@signalapp/libsignal-client/dist/Errors.js:39:19)"}
{"level":50,"time":"2023-03-25T18:05:44.196Z","msg":"MessageReceiver.decryptAndCacheBatch error when processing the envelope LibSignalError: session with [REDACTED]2ce.1 not found\n    at new LibSignalErrorBase ([REDACTED]/resources/app.asar/node_modules/@signalapp/libsignal-client/dist/Errors.js:39:19)"}
{"level":30,"time":"2023-03-25T18:05:44.196Z","msg":"MessageReceiver.decryptAndCacheBatch storing 0 decrypted envelopes, keeping 1 failed envelopes."}
{"level":30,"time":"2023-03-25T18:05:44.196Z","msg":"commitZoneChanges(MessageReceiver): pending sender keys 0, pending sessions 0, pending unprocessed 1"}
```

--- a/libsignal-service/src/cipher.rs
+++ b/libsignal-service/src/cipher.rs
@@ -164,6 +164,20 @@ where
                 )?;
                 Plaintext { metadata, data }
             },
+            Type::PlaintextContent => {
+                log::warn!("Envelope with plaintext content.  This usually indicates a decryption retry.");
+                let metadata = Metadata {
+                    sender: envelope.source_address(),
+                    sender_device: envelope.source_device(),
+                    timestamp: envelope.server_timestamp(),
+                    needs_receipt: false,
+                    unidentified_sender: false,
+                };
+                Plaintext {
+                    metadata,
+                    data: ciphertext.clone(),
+                }
+            },
             Type::Ciphertext => {
                 let sender = get_preferred_protocol_address(
                     &self.session_store,

